tests/unit/test_client.py

```diff
@@ -1368,6 +1368,23 @@ def return_mock_tuple(**kwargs):
         service_client.test_operation(Foo='one')
         self.assertFalse(self.endpoint.make_request.called)
 
+    def test_getattr_emits_event(self):
+        emitter = mock.Mock()
+        emitter.emit_until_response.return_value = (None, None)
+
+        creator = self.create_client_creator(event_emitter=emitter)
+        service_client = creator.create_client('myservice', 'us-west-2')
+
+        emitter.emit_until_response.assert_not_called()
+
+        with self.assertRaises(AttributeError):
+            service_client.attribute_that_does_not_exist
+
+        emitter.emit_until_response.assert_called_once_with(
+            'getattr.myservice.attribute_that_does_not_exist',
+            client=service_client
+        )
+
```

Comment:
We should have a unit test here that asserts that an event handler that returns a response will be used instead of raising an attribute error (i.e a positive test case).